src/main/java/frc/robot/subsystems/shooter/ShooterIOTalonFX.java

```diff
@@ -0,0 +1,294 @@
+      drumMotorSupplyCurrentStatusSignal,
+      dunkerMotorSupplyCurrentStatusSignal,
+      angleMotorSupplyCurrentStatusSignal
+    );
+    
+    // Updates Left Shooter Motor Inputs
+    shooterInputs.shootMotorLeftStatorCurrentAmps = shootMotorLeftStatorCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.shootMotorLeftSupplyCurrentAmps = shootMotorLeftSupplyCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.shootMotorLeftVelocityRPS = shootMotorLeftVelocityStatusSignal.getValueAsDouble();
+
+    // Updates Right Shooter Motor Inputs
+    shooterInputs.shootMotorRightStatorCurrentAmps = shootMotorRightStatorCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.shootMotorRightSupplyCurrentAmps = shootMotorRightSupplyCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.shootMotorRightVelocityRPS = shootMotorRightVelocityStatusSignal.getValueAsDouble();
+
+    // Updates Angle Motor Inputs
+    shooterInputs.angleMotorStatorCurrentAmps = angleMotorStatorCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.angleMotorSupplyCurrentAmps = angleMotorSupplyCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.angleMotorVelocityRPS = angleMotor.get();
+
+    // Updates Drum Motor Inputs
+    shooterInputs.drumMotorStatorCurrentAmps = drumMotorStatorCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.drumMotorSupplyCurrentAmps = drumMotorSupplyCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.drumMotorVelocityRPS = drumMotorVelocityStatusSignal.getValueAsDouble();
+
+    // Updates Dunker Motor Inputs
+    shooterInputs.dunkerMotorStatorCurrentAmps = dunkerMotorStatorCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.dunkerMotorSupplyCurrentAmps = dunkerMotorSupplyCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.dunkerMotorVelocityRPS = dunkerMotorVelocityStatusSignal.getValueAsDouble();
+
+    // Updates Shooter Sensor
+    shooterInputs.isShooterSensorBlocked = shooterSensor.get();    
+
+  }
+
+
+  @Override
+  public void setDrumMotorVelocity(double rps) {
+    drumMotor.setControl(drumMotorVelocityRequest.withVelocity(rps));
+    this.drumMotorRequestedVelocityRPS = rps;
+  }
+
+
+  @Override
+  public void setDunkerMotorVelocity(double rps) {
+    dunkerMotor.setControl(dunkerMotorVelocityRequest.withVelocity(rps));
+    this.dunkerMotorVelocityRPS = rps;
+  }
+
+
+  @Override
+  public void setShooterMotorLeftVelocity(double rps) {
+    shootMotorLeft.setControl(shootMotorLeftVelocityRequest.withVelocity(rps));
+    this.shootMotorLeftRequestedVelocityRPS = rps;
+  }
+
+
+  @Override
+  public void setShooterMotorRightVelocity(double rps) {
+    shootMotorRight.setControl(shootMotorRightVelocityRequest.withVelocity(rps));
+    this.shootMotorRightRequestedVelocityRPS = rps;
+  }
+
+
+  @Override
+  public void setAngle(double angle) {
+    // Magic Motion
+  }  
+
+
+  private void configShootMotors(int leftMotorID, int rightMotorID){
```

Comment:
I'm not sure if two devices can share the same config objects. I would recommend creating both TalonFX objects in the constructor and passing one at a time as a parameter to this method (and invoking this method twice, once for each object).